build/webpack.system.conf.js

```diff
@@ -1,97 +0,0 @@
-"use strict"
-const utils = require("./utils")
-const webpack = require("webpack")
-const path = require("path")
-const config = require("../config")
-const merge = require("webpack-merge")
-const baseWebpackConfig = require("./webpack.base.conf")
-const MergeWebpackPlugin = require("webpack-merge-and-include-globally")
-const MiniCssExtractPlugin = require("mini-css-extract-plugin")
-const OptimizeCSSPlugin = require("optimize-css-assets-webpack-plugin")
-const CopyWebpackPlugin = require("copy-webpack-plugin")
-const SafeParser = require("postcss-safe-parser")
-
-const env = require("../config/prod.env")
-
-baseWebpackConfig.entry = {
-  system: ["./src/system.js"],
-}
-
-const webpackConfig = merge(baseWebpackConfig, {
-  module: {
-    rules: utils.styleLoaders({
-      sourceMap: config.system.productionSourceMap,
-      extract: true,
-      usePostCSS: true,
-    }),
-  },
-  devtool: config.build.productionSourceMap ? config.system.devtool : false,
-  output: {
-    path: config.system.assetsRoot,
-    filename: utils.assetsSystemPath("[name].js"),
-    library: "[name]",
-    libraryTarget: config.system.libraryTarget,
-  },
-  performance: {
-    hints: config.system.performanceHints,
-  },
-  plugins: [
-    // http://vuejs.github.io/vue-loader/en/workflow/production.html
-    new webpack.DefinePlugin({
-      "process.env": env,
-    }),
-    // extract css into its own file
-    new MiniCssExtractPlugin({
-      filename: utils.assetsSystemPath("[name].css"),
-    }),
-    // Compress extracted CSS. We are using this plugin so that possible
-    // duplicated CSS from different components can be deduped.
-    new OptimizeCSSPlugin({
-      cssProcessorOptions: { parser: SafeParser },
-    }),
-    // keep module.id stable when vendor modules does not change
-    new webpack.HashedModuleIdsPlugin(),
-    // enable scope hoisting
-    new webpack.optimize.ModuleConcatenationPlugin(),
-    // Copy and merge Sass tokens and system utilities as well
-    new MergeWebpackPlugin({
-      files: {
-        [utils.assetsSystemPath("system.utils.scss")]: [
-          "./src/assets/tokens/tokens.scss",
-          "./src/styles/_spacing.scss",
-          "./src/styles/_mixins.scss",
-          "./src/styles/_functions.scss",
```

Comment:
What are your thoughts on preserving some of the build configuration somewhere else? Will that be possible? Don’t think all of this can be just removed.

Reply:
It could be set these customizations of the build process in multiple places. 

We could have a vueds.config.js or we could include it as [pluginSettings](https://cli.vuejs.org/dev-guide/plugin-dev.html#custom-options-for-3rd-party-plugins) directly in vue.config.js.

I would recommend though not to multiply the customization entry points. Every customization is source of bugs and misunderstandings. It can make the developers codebase very ugly and vue-design-systems changes incredibly more dangerous. Plus it is more documentation work ;).

Reply:
I do agree.